rasterio/transform.py

```diff
@@ -20,12 +20,10 @@ def guard_transform(transform):
     """Return an Affine transformation instance."""
     if not isinstance(transform, Affine):
         if tastes_like_gdal(transform):
-            warnings.warn(
-                "GDAL-style transforms are deprecated and will not "
-                "be supported in Rasterio 1.0.",
-                FutureWarning,
-                stacklevel=2)
-            transform = Affine.from_gdal(*transform)
+            raise ValueError(
+                "GDAL-style transforms have been deprecated in 1.0.  This "
```

Comment:
@geowurster I think you can leave off the `in 1.0`; there could be other releases between now and then, and the important part is that this was deprecated prior to 1.0